### PR TITLE
Add methods for get or update instantiate_model

### PR DESCRIPTION
**Pull request by itdependsnetworks**

### CODE REVIEW

**Comment by itdependsnetworks** [comment]
Doh’, this doesn’t make sense. When I went to use this, I realized the method was conditionally adding. This is just redundant

**Comment by itdependsnetworks** [comment]
Double Doh'!! ok, red herring, there was an issue, I didn't see where the code was actually issuing a `self.add` (it was nested more), so I thought it was adding.... it was :)

File: diffsync/store/__init__.py [hunk 1]
```diff
@@ -159,6 +159,24 @@ def get_or_instantiate(
 
         return obj, created
 
+    def get_or_instantiate_model(self, obj: "DiffSyncModel") -> Tuple["DiffSyncModel", bool]:
```
Comment (Kircheneer) [inline diffsync/store/__init__.py:162]: Maybe this should be called `get_or_add_model_instance`, seeing as
- the model class instance is already instantiated
- the model itself is already known to the `DiffSync` instance at this point (i.e. we aren't "adding" a model)
- "`get_or_add`" also accurately reflects the methods called against `self`

File: diffsync/store/__init__.py [hunk 2]
```diff
@@ -159,6 +159,24 @@ def get_or_instantiate(
...
+        """Attempt to get the object with provided obj identifiers or instantiate obj.
+
+        Args:
+            obj: An obj of the DiffSyncModel to get or create.
+
+        Returns:
+            Provides the existing or new object and whether it was created or not.
```
Comment (Kircheneer) [inline diffsync/store/__init__.py:169]: This needs to be updated to reflect the functionality (created vs. instantiated vs. added).

File: diffsync/store/__init__.py [hunk 3]
```diff
@@ -188,6 +206,34 @@ def update_or_instantiate(
...
 
+    def update_or_instantiate_model(self, obj: "DiffSyncModel") -> Tuple["DiffSyncModel", bool]:
+        """Attempt to update an existing object with provided ids/attrs or instantiate obj.
+
+        Args:
+            instance: An instance of the DiffSyncModel to update or create.
+
+        Returns:
+            Provides the existing or new object and whether it was created or not.
+        """
```
Comment (Kircheneer) [inline diffsync/store/__init__.py:217]: See comments for current `get_or_instantiate_model` method

File: diffsync/store/__init__.py [hunk 4]
```diff
@@ -188,6 +206,34 @@ def update_or_instantiate(
...
+        model = obj.get_type()
+        ids = obj.get_unique_id()
+        attrs = obj.get_attrs()
+
+        created = False
+        try:
+            obj = self.get(model=model, identifier=ids)
+        except ObjectNotFound:
+            # Add the object to diffsync adapter
```
Comment (Kircheneer) [inline diffsync/store/__init__.py:226]: I believe the concept of an adapter is something that the SSoT framework uses, but it is never mentioned in this repository. Possibly:
`# Add the object to the diffsync instance`?

File: diffsync/store/__init__.py [hunk 5]
```diff
@@ -188,6 +206,34 @@ def update_or_instantiate(
...
+            self.add(obj=obj)
+            created = True
+
+        # Update existing obj with attrs
+        for attr, value in attrs.items():
+            if getattr(obj, attr) != value:
```
Comment (Kircheneer) [inline diffsync/store/__init__.py:232]: I believe we can omit this check to make it simpler. Just `setattr` always, if its the same the outcome is also the same.

**Review by Kircheneer — CHANGES_REQUESTED** [review]

File: diffsync/store/__init__.py [hunk 6]
```diff
@@ -188,6 +206,33 @@ def update_or_instantiate(
 
         return obj, created
 
+    def update_or_add_model_instance(self, obj: "DiffSyncModel") -> Tuple["DiffSyncModel", bool]:
+        """Attempt to update an existing object with provided ids/attrs or instantiate obj.
+
+        Args:
+            instance: An instance of the DiffSyncModel to update or create.
```
Comment (Kircheneer) [inline diffsync/store/__init__.py:213]:
```suggestion
            instance: An instance of the DiffSyncModel to update or add.
```

**Review by chadell — APPROVED** [review]
LGTM

**Review by Kircheneer — APPROVED** [review]

**Pull request merged**